### ISSUE TITLE
Add method that creates appropriate RDF resource type based on the input string

### ISSUE THREAD

**Issue by sbpcs59** [ropensci/redland-bindings#14]
The Node class can create three types of nodes:

    Node(uri="http://example.com/thing")  : creates a node of type "uri"
    Node(litteral="10.2") creates a node of type "literal"
    Node("dog") creates a node of type "literal"
    Node() creates a blank node

The calling program has to know what type of node to create to use these arguments.

It would be very useful to have a utility method that accepts strings for the RDF 
subject, predicate and object and determines whether each is a URI, a literal or a
blank node, for example:

This method could take advantage of the following restrictions from the RDF model
(http://www.w3.org/TR/rdf11-concepts/#section-triples) where:

       a subject can be a URI or a blank node
       a predicate must be a URI
       an object can be a URI, litteral or blank node

One way to do this is to implement this in the Statement class, i.e:

   createStatement(stmt, subject="http://www.example.com/thing",
                   predicate="http://www.example.com/occurredAt",
                   object="2015-01-01T10:22:01")

This would create a Statement with a URI subject, URI predicate and litteral object.

**Comment by mbjones** [comment]
I like your approach.  An improvement might be to exploit the restrictions such that:

- subject argument passed as:
    - string (non-empty): then convert it to a uri and create a Node with the URI
    - NULL: then make a blank node
- predicate always passed as string and converted to a Node
- object
    - NULL: create a blank node
    - string: create a literal node
    - string with additional 'objectIsUri=true` argument, then create as a URI node

So then, some examples would be:
```r
createStatement(stmt, subject="http://www.example.com/thing", predicate="http://www.example.com/occurredAt", object="2015-01-01T10:22:01")
createStatement(stmt, subject=NULL, predicate="http://www.example.com/occurredAt", object="2015-01-01T10:22:01")
createStatement(stmt, subject="http://www.example.com/thing", predicate="http://www.example.com/occurredAt", object="timeuri:2015-01-01T10:22:01", objectIsUri=true)
```

What do you think?

**Comment by sbpcs59** [comment]
This looks good. 

Should support be added for blank node identifiers, i.e. "_:somestring" , which are not an official part of the RDF syntax: http://www.w3.org/TR/rdf11-concepts/#dfn-blank-node-identifier? If yes, then
we need to explicitly state that a blank node is being created, not a literal node.

**Comment by mbjones** [comment]
Yeah, we do need to support blank node identifiers. But wouldn't we need to ensure that any blank node identifiers are unique in the document?  Shouldn't we auth-generate them when a blank node is created?  I guess there could be a use case for the user to set the blank node id, but then they would have to ensure the proper level of uniqueness. I used the redlands python bindings the other day to create that geolink set of triples that I showed you, and I found that creating blank nodes was a bit awkward, which is why I created all of those UUIDs. 

It would be nice if the signature didn't have too many arguments to confuse things.  Would it be reasonable to assume that a string matching the regex `_:.*` should be a blank node? If so, then we wouldn't need another parameter for the subject or object.

**Comment by sbpcs59** [comment]
Creating a blank node when "_:.*" is a good idea for the createStatement method for arguments.
subject="_:<string>", object="_:<string>".

I'd also like to add this to the Node class, i.e. node <- new("Node", world, "_:somthing") to create
a blank node, as new("Node", world, literal="_:something") could always be used to explicitly
create a literal.

Regarding auto generating node ids: redland will create a blank node and assign an internally generated nodeId when librdr_new_node(world) is called. This is currently done when new("Node", "") is called.
This nodeId is available by calling librdf_node_get_blank_identifier().  So all that would need to be added is a method to Node.R to call librdr_node_get_blank_identifier.

**Comment by mbjones** [comment]
@sbpcs59 Perfect.

**Comment by sbpcs59** [comment]
Implemented in commit 19eeec7d8816f8e3c100e859a5c07c8b1420d98e . Changes were made to
Node.initialize() and Statement.initialize() to implement this.
